### PR TITLE
Param utils

### PR DESCRIPTION
**Pull request by harshit-gupta-cor**
This PR follows the discussion in #20. The print statements will be replaced with loggers once the full pipeline is on the repo.

### CODE REVIEW

File: simSPI/linear_simulator/params_utils.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+            dictionary containing:
+            "defocus_u": torch.Tensor
+                Tensor of size (batch_size,1,1,1) that contains the major
+                defocus value in microns
+            "defocus_v": torch.Tensor
+                Tensor of size (batch_size,1,1,1) that contains the minor
+                defocus value in microns
+            "defocus_angle": torch.Tensor
+                Tensor of size (batch_size,1,1,1) that contains the major
+                defocus value in microns
```
Comment (geoffwoollard) [inline simSPI/linear_simulator/params_utils.py:174]: typo. units of angle are radians?

File: simSPI/linear_simulator/params_utils.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+        """Get the parameters for the shift of the particle from the starfile.
+
+        If config.shift is True else returns {}
+
+        Returns
+        -------
+            shift_params: dict of type str to {torch.Tensor}
+            dictionary containing
+            'shift_x': torch.Tensor (batch_size,)
+                batch of shifts along horizontal axis
```
Comment (geoffwoollard) [inline simSPI/linear_simulator/params_utils.py:210]: put in the units. pixels, angstroms, etc?

File: simSPI/linear_simulator/params_utils.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+            'shift_y': torch.Tensor (batch_size,)
+                batch of shifts along vertical axis
```
Comment (geoffwoollard) [inline simSPI/linear_simulator/params_utils.py:212]: units

File: simSPI/linear_simulator/params_utils.py [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+            dictionary containing:
+            "defocus_u": torch.Tensor
+                Tensor of size (batch_size,1,1,1) that contains the major
+                defocus value in microns
+            "defocus_v": torch.Tensor
+                Tensor of size (batch_size,1,1,1) that contains the minor
+                defocus value in microns
+            "defocus_angle": torch.Tensor
+                Tensor of size (batch_size,1,1,1) that contains the major
+                defocus value in microns
```
Comment (geoffwoollard) [inline simSPI/linear_simulator/params_utils.py:310]: angle units

File: simSPI/linear_simulator/params_utils.py [hunk 5]
```diff
@@ -0,0 +1,361 @@
+"""contains functions and classes for parameter generation."""
+from abc import ABCMeta, abstractstaticmethod
+
+import numpy as np
+import starfile
+import torch
+import torch.fft
```
Comment (geoffwoollard) [inline simSPI/linear_simulator/params_utils.py:7]: did you use this?

**Review by geoffwoollard — COMMENTED** [review]
Some small changes with unit documentation

File: simSPI/linear_simulator/params_utils.py [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+        Class containing parameters of the dataset generator.
+
+    Returns
+    -------
+    config: class
+    """
+    config.starfile_available = config.input_starfile_path != ""
+    if config.starfile_available:
+        config = starfile_opticsparams(config)
+    config.ctf_size = config.side_len  # TODO: make it adaptable
```
Comment (geoffwoollard) [inline simSPI/linear_simulator/params_utils.py:32]: Make a github issue and take out the TODO
https://deepsource.io/gh/compSPI/simSPI/run/e173e44e-f4de-48fb-bedf-1e4c1fe17ad9/python/PYL-W0511

File: simSPI/linear_simulator/params_utils.py (hunk 1 above)
Comment (harshit-gupta-cor) [reply simSPI/linear_simulator/params_utils.py]: Done. Thanks!

File: simSPI/linear_simulator/params_utils.py (hunk 2 above)
Comment (harshit-gupta-cor) [reply simSPI/linear_simulator/params_utils.py]: Added now. Thanks

File: simSPI/linear_simulator/params_utils.py (hunk 3 above)
Comment (harshit-gupta-cor) [reply simSPI/linear_simulator/params_utils.py]: Done

File: simSPI/linear_simulator/params_utils.py (hunk 4 above)
Comment (harshit-gupta-cor) [reply simSPI/linear_simulator/params_utils.py]: Done

File: simSPI/linear_simulator/params_utils.py (hunk 5 above)
Comment (harshit-gupta-cor) [reply simSPI/linear_simulator/params_utils.py]: Removed

File: simSPI/linear_simulator/params_utils.py (hunk 6 above)
Comment (harshit-gupta-cor) [reply simSPI/linear_simulator/params_utils.py]: Removed the TODO. It was some remnant from past.

**Comment by geoffwoollard** [comment]
You should resolve the failed test

**Comment by harshit-gupta-cor** [comment]
This PR has been replaced by #74. There were some merge issues.

**Pull request closed without merging**